hedera-node/src/main/java/com/hedera/services/store/tokens/unique/UniqueTokenStore.java

```diff
@@ -66,34 +73,77 @@ public UniqueTokenStore(final EntityIdSource ids,
 	) {
 		super(ids, validator, properties, tokens, tokenRelsLedger);
 		this.uniqueTokenSupplier = uniqueTokenSupplier;
+		this.lastMintedSerialNumbers = new ArrayList<>(); // TODO MAX_SIZE, probably 50
 	}
 
 	@Override
-	public ResponseCodeEnum mint(final TokenID tId, final String memo, final RichInstant creationTime) {
-		return tokenSanityCheck(tId, (merkleToken -> {
+	public List<Long> popLastMintedSerialNumbers() {
+		var copy = new ArrayList<>(lastMintedSerialNumbers);
+		lastMintedSerialNumbers.clear();
+		return copy;
+	}
+
+	@Override
+	public ResponseCodeEnum wipe(final AccountID aId, final TokenID tId, final long wipingAmount, final boolean skipKeyCheck) {
+		return null;
+	}
+
+	@Override
+	public ResponseCodeEnum mint(final TokenMintTransactionBody txBody, final RichInstant creationTime) {
+		var provisionalUniqueTokens = new ArrayList<Pair<MerkleUniqueTokenId, MerkleUniqueToken>>();
+		var tokenId = txBody.getToken();
+		return tokenSanityCheck(tokenId, merkleToken -> {
+
```

Comment:
We have to verify that the token is of type `NON_FUNGIBLE_UNIQUE`. If not we should throw `INVALID_TRANSACTION_BODY`. We should have the equivalent check for fungible `mints`